src/System.Memory/tests/ArrayBufferWriter/ArrayBufferWriterTests.T.cs

```diff
@@ -0,0 +1,401 @@
+                ReadOnlyMemory<T> previousMemory = output.WrittenMemory;
+                ReadOnlySpan<T> previousSpan = output.WrittenSpan;
+                Assert.True(previousSpan.SequenceEqual(previousMemory.Span));
+                output.Advance(10);
+                Assert.False(previousMemory.Span.SequenceEqual(output.WrittenMemory.Span));
+                Assert.False(previousSpan.SequenceEqual(output.WrittenSpan));
+                Assert.True(output.WrittenSpan.SequenceEqual(output.WrittenMemory.Span));
+            }
+
+            {
+                var output = new ArrayBufferWriter<T>();
+                _ = output.GetSpan(20);
+                WriteData(output, 10);
+                ReadOnlyMemory<T> previousMemory = output.WrittenMemory;
+                ReadOnlySpan<T> previousSpan = output.WrittenSpan;
+                Assert.True(previousSpan.SequenceEqual(previousMemory.Span));
+                Assert.Throws<InvalidOperationException>(() => output.Advance(15));
+                output.Advance(10);
+                Assert.False(previousMemory.Span.SequenceEqual(output.WrittenMemory.Span));
+                Assert.False(previousSpan.SequenceEqual(output.WrittenSpan));
+                Assert.True(output.WrittenSpan.SequenceEqual(output.WrittenMemory.Span));
+            }
+        }
+
+        [Fact]
+        public void AdvanceZero()
+        {
+            var output = new ArrayBufferWriter<T>();
+            WriteData(output, 2);
+            Assert.Equal(2, output.WrittenCount);
+            ReadOnlyMemory<T> previousMemory = output.WrittenMemory;
+            ReadOnlySpan<T> previousSpan = output.WrittenSpan;
+            Assert.True(previousSpan.SequenceEqual(previousMemory.Span));
+            output.Advance(0);
+            Assert.Equal(2, output.WrittenCount);
+            Assert.True(previousMemory.Span.SequenceEqual(output.WrittenMemory.Span));
+            Assert.True(previousSpan.SequenceEqual(output.WrittenSpan));
+            Assert.True(output.WrittenSpan.SequenceEqual(output.WrittenMemory.Span));
+        }
+
+        [Fact]
+        public void InvalidAdvance()
+        {
+            {
+                var output = new ArrayBufferWriter<T>();
+                Assert.Throws<ArgumentException>(() => output.Advance(-1));
+                Assert.Throws<InvalidOperationException>(() => output.Advance(output.Capacity + 1));
+            }
+
+            {
+                var output = new ArrayBufferWriter<T>();
+                WriteData(output, 100);
+                Assert.Throws<InvalidOperationException>(() => output.Advance(output.FreeCapacity + 1));
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(SizeHints))]
+        public void GetSpan_DefaultCtor(int sizeHint)
+        {
+            {
+                var output = new ArrayBufferWriter<T>();
+                Span<T> span = output.GetSpan();
+                Assert.Equal(1, span.Length);
+            }
+
+            {
```

Comment:
Most of these scoped tests seem like they should be split into two named methods.  This one in particular, since the top half doesn't use the input parameter.